File: miniapp/src/services/checklist.ts
import { request } from './api'

export type ChecklistCategory =
  | 'document'
  | 'electronics'
  | 'clothing'
  | 'medicine'
  | 'food'
  | 'home'
  | 'pet'
  | 'task'
  | 'other'

export type ChecklistSource = 'template' | 'manual' | 'ai_generated'

export interface ChecklistTemplate {
  id: string
  label: string
  category: ChecklistCategory
  sortOrder: number
  isDefault: boolean
}

export interface ChecklistItem {
  id: string
  tripId: string
  label: string
  checked: boolean
  category: ChecklistCategory
  source: ChecklistSource
  templateId: string | null
  sortOrder: number
  createdAt: string
  updatedAt: string
}

export interface ChecklistItemCreate {
  label: string
  category?: ChecklistCategory
  sortOrder?: number
  templateId?: string
  checked?: boolean
}

export interface ChecklistItemPatch {
  label?: string
  category?: ChecklistCategory
  sortOrder?: number
  checked?: boolean
}

export const CATEGORY_LABELS: Record<ChecklistCategory, string> = {
  document: '证件',
  electronics: '电子',
  clothing: '服饰',
  medicine: '药品',
  food: '食品',
  home: '家居',
  pet: '宠物',
  task: '事务',
  other: '其他',
}

export const checklistApi = {
  templates: () => request<ChecklistTemplate[]>('/checklist-templates'),
  list: (tripId: string) =>
    request<ChecklistItem[]>(`/trips/${tripId}/checklist-items`),
  create: (tripId: string, payload: ChecklistItemCreate) =>
    request<ChecklistItem>(`/trips/${tripId}/checklist-items`, {
      method: 'POST',
      data: payload,
    }),
  patch: (id: string, payload: ChecklistItemPatch) =>
    request<ChecklistItem>(`/checklist-items/${id}`, {
      method: 'PATCH',
      data: payload,
    }),
  delete: (id: string) =>
    request<void>(`/checklist-items/${id}`, { method: 'DELETE' }),
}

File: backend/app/schemas/checklist.py
from datetime import datetime
from enum import StrEnum
from uuid import UUID

from pydantic import BaseModel, ConfigDict, Field


class ChecklistCategory(StrEnum):
    document = "document"
    electronics = "electronics"
    clothing = "clothing"
    medicine = "medicine"
    food = "food"
    home = "home"
    pet = "pet"
    task = "task"
    other = "other"


class ChecklistSource(StrEnum):
    template = "template"
    manual = "manual"
    ai_generated = "ai_generated"


_camel = ConfigDict(populate_by_name=True, from_attributes=True)


class ChecklistTemplateOut(BaseModel):
    model_config = _camel

    id: UUID
    label: str
    category: ChecklistCategory
    sort_order: int = Field(serialization_alias="sortOrder")
    is_default: bool = Field(serialization_alias="isDefault")


class ChecklistItemCreate(BaseModel):
    model_config = _camel

    label: str = Field(min_length=1, max_length=64)
    category: ChecklistCategory = ChecklistCategory.other
    sort_order: int = Field(default=0, validation_alias="sortOrder")
    template_id: UUID | None = Field(default=None, validation_alias="templateId")
    checked: bool = False


class ChecklistItemPatch(BaseModel):
    model_config = _camel

    label: str | None = Field(default=None, min_length=1, max_length=64)
    category: ChecklistCategory | None = None
    sort_order: int | None = Field(default=None, validation_alias="sortOrder")
    checked: bool | None = None


class ChecklistItemOut(BaseModel):
    model_config = _camel

    id: UUID
    trip_id: UUID = Field(serialization_alias="tripId")
    label: str
    checked: bool
    category: ChecklistCategory
    source: ChecklistSource
    template_id: UUID | None = Field(serialization_alias="templateId")
    sort_order: int = Field(serialization_alias="sortOrder")
    created_at: datetime = Field(serialization_alias="createdAt")
    updated_at: datetime = Field(serialization_alias="updatedAt")
